oap-server/server-alarm-plugin/src/test/java/org/apache/skywalking/oap/server/core/alarm/provider/CompositeRuleEvaluateTest.java

```diff
@@ -0,0 +1,151 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.core.alarm.provider;
+
+import org.apache.skywalking.oap.server.core.alarm.AlarmMessage;
+import org.apache.skywalking.oap.server.core.alarm.provider.expression.Expression;
+import org.apache.skywalking.oap.server.core.alarm.provider.expression.ExpressionContext;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public class CompositeRuleEvaluateTest {
+
+    private CompositeRuleEvaluate ruleEvaluate;
+
+    @Before
+    public void init() {
+        Expression expression = new Expression(new ExpressionContext());
+        ruleEvaluate = new CompositeRuleEvaluate(expression);
+    }
+
+    @Test
+    public void testEvaluateMessageWithAndOp() {
+        List<CompositeAlarmRule> compositeAlarmRules = new ArrayList<>();
+        CompositeAlarmRule compositeAlarmRule = new CompositeAlarmRule("dummy", "a_rule && b_rule", "composite rule triggered!");
+        compositeAlarmRules.add(compositeAlarmRule);
+        List<AlarmMessage> alarmMessages = getAlarmMessages();
+        List<AlarmMessage> compositeMsgs = ruleEvaluate.evaluate(compositeAlarmRules, alarmMessages);
+        assertThat(compositeMsgs.size(), is(1));
+        assertThat(compositeMsgs.get(0).getAlarmMessage(), is("composite rule triggered!"));
+        assertThat(compositeMsgs.get(0).getRuleName(), is("dummy"));
+        assertThat(compositeMsgs.get(0).getId0(), is("id0"));
+        assertThat(compositeMsgs.get(0).getId1(), is("id1"));
+        assertThat(compositeMsgs.get(0).isOnlyAsCondition(), is(false));
+    }
+
+    @Test
+    public void testEvaluateMessageWithNotExistsRule() {
+        List<CompositeAlarmRule> compositeAlarmRules = new ArrayList<>();
+        CompositeAlarmRule compositeAlarmRule = new CompositeAlarmRule("dummy", "a_rule && not_exist_rule", "composite rule triggered!");
+        compositeAlarmRules.add(compositeAlarmRule);
+        List<AlarmMessage> alarmMessages = getAlarmMessages();
+        List<AlarmMessage> compositeMsgs = ruleEvaluate.evaluate(compositeAlarmRules, alarmMessages);
```

Comment:
Just for confirmation, at here, nothing will show up as log or exception, right? As `not_exist_rule` doesn't exist.

Reply:
Yes, the error log is in Expression.eval method